ml/src/main/scala/org/apache/spark/ml/h2o/models/H2OMOJOModel.scala

```diff
@@ -211,11 +217,38 @@ class H2OMOJOModel(override val uid: String) extends H2OMOJOModelBase[H2OMOJOMod
 
 object H2OMOJOModel extends H2OMOJOReadable[PyH2OMOJOModel] with H2OMOJOLoader[PyH2OMOJOModel] {
 
+  private def removeMetaField(json: JsonElement): JsonElement = {
+    if (json.isJsonObject) {
+      json.getAsJsonObject.remove("__meta")
+      json.getAsJsonObject.entrySet().asScala.map(_.getValue).foreach(removeMetaField)
+    }
+    if (json.isJsonArray) {
+      json.getAsJsonArray.asScala.foreach(removeMetaField)
+    }
+    json
+  }
+
+
+  private def getModelDetails(mojoData: Array[Byte]): String = {
+    val is = new ByteArrayInputStream(mojoData)
+    val reader = MojoReaderBackendFactory.createReaderBackend(is, MojoReaderBackendFactory.CachingStrategy.MEMORY)
```

Comment:
Just an idea... :-) WDYT about doing the following?
* move these two lines to `createFromMojo` 
* call `ModelMojoReader.readFrom(reader)` in `createFromMojo` instead of `ModelSerializationSupport.getMojoModel(mojoData)`
* pass the `reader` to `getModelDetails` instead of `mojoData`

Reply:
Perfect idea! I had it like this originally as I wanted to reuse `reader`. However the method `ModelMojoReader.readFrom(reader)` has ugly side effect of closing the reader so we had to create a new reader